### DISCUSSION TITLE
Can I Change the Teleport log timezone?

### DISCUSSION THREAD

**Discussion by mikeD106** [gravitational/teleport#8366] [Q&A]
Hello guys, I noticed that Teleport wrote the log use UTC+0 timezone, like this

{"ei":0,"event":"role.created","uid":"62a2e861-2146-452d-b4a5-c14de4ab0d94","code":"T9000I","time":"2021-09-24T09:00:36.614Z","cluster_name":"clusterManager","name":"admin","expires":"0001-01-01T00:00:00Z","user":"system"}
{"ei":0,"event":"role.created","uid":"606d139c-e437-4450-a0e1-0eaec6160e0e","code":"T9000I","time":"2021-09-24T09:00:50.397Z","cluster_name":"clusterManager","name":"Example1","expires":"0001-01-01T00:00:00Z","user":"cca17864-ca39-472b-8e8c-e4ca5c1c4157.clusterManager"}
{"ei":0,"event":"role.created","uid":"daf2452a-ff94-4657-a443-dc6d0b11bc83","code":"T9000I","time":"2021-09-24T09:00:55.933Z","cluster_name":"clusterManager","name":"Example2","expires":"0001-01-01T00:00:00Z","user":"cca17864-ca39-472b-8e8c-e4ca5c1c4157.clusterManager"}
{"ei":0,"event":"role.created","uid":"39eb4fcc-6a13-477e-8ef4-a33820fc8987","code":"T9000I","time":"2021-09-24T09:00:58.819Z","cluster_name":"clusterManager","name":"Example3","expires":"0001-01-01T00:00:00Z","user":"cca17864-ca39-472b-8e8c-e4ca5c1c4157.clusterManager"}
{"ei":0,"event":"user.create","uid":"a9d93415-dfab-462c-8bf8-faca942de6c2","code":"T1002I","time":"2021-09-24T09:03:18.573Z","cluster_name":"clusterManager","user":"cca17864-ca39-472b-8e8c-e4ca5c1c4157.clusterManager","name":"michael.dragon","expires":"0001-01-01T00:00:00Z","roles":["Example1"],"connector":"local"}
{"ei":0,"event":"reset_password_token.create","uid":"93d791ba-5b98-4272-b860-de82050f07d1","code":"T6000I","time":"2021-09-24T09:03:18.686Z","cluster_name":"clusterManager","name":"usernameExample","expires":"2021-09-24T10:03:18.686007069Z","ttl":"1h0m0s","user":"cca17864-ca39-472b-8e8c-e4ca5c1c4157.clusterManager"}
{"ei":0,"event":"user.login","uid":"0ff08df7-3a7c-43a9-9234-a5a09af1bd88","code":"T1000I","time":"2021-09-24T09:34:51.743Z","cluster_name":"clusterManager","user":"usernameExample","success":true,"method":"local","mfa_device":{"mfa_device_name":"otp","mfa_device_uuid":"7063f2b9-8428-43ee-a202-dec465a9941d","mfa_device_type":"otp"}}

and after search the documentation on Teleport official website and Discussion on Teleport official github, I didn't come across anything that related to how to change the timezone on the log file. the question is can I change any config about the timezone for Teleport log?

thank you for your attention and help. Much Appreciate it!

**Comment by stevenGravy** [comment]
Hi @mikeD106 set the variable `TZ` to your particular zone/

`export TZ=America/New_York`

**Reply by mikeD106** [reply]
thanks @stevenGravy for your help, I really appreciate it!

**Reply by mikeD106** [reply]
sorry for another follow up question, so what I need is only run the 'export TZ=America/New_York' command before I install teleport or is there something I need to change in teleport config?

**Reply by stevenGravy** [reply]
that's a environment variable.  If you're running by systemd just add that as a environment variable. Here's an example https://serverfault.com/questions/413397/how-to-set-environment-variable-in-systemd-service

**Reply by stevenGravy** [reply]
The teleport systemd https://goteleport.com/docs/setup/admin/daemon/

**Reply by mikeD106** [reply]
according to the link you gave me, so my teleport.service for systemd should be like this, is that right?

```
[Unit]
Description=Teleport SSH Service
After=network.target

[Service]
Environment="TZ=Asia/Singapore"
Type=simple
Restart=on-failure
ExecStart=/usr/local/bin/teleport start --pid-file=/run/teleport.pid --config=/etc/teleport.yaml
ExecReload=/bin/kill -HUP $MAINPID
PIDFile=/run/teleport.pid
LimitNOFILE=8192

[Install]
WantedBy=multi-user.target
```

because after I added the TZ environment and tried to run with "systemctl restart teleport.service" or even tried reinstall teleport with added TZ environment inside teleport.service file, the log in /var/lib/teleport/log/events.log still written in UTC+0

**Comment by webvictim** [comment]
My understanding is that Teleport deliberately always logs all events in the event log as UTC to avoid timezone issues - this is what the `Z` at the end of the timestamp means. I don't believe it's possible to change this.
